src/components/program/ProgramContentMenu.tsx

```diff
@@ -279,6 +294,44 @@ const SortBySectionItem: React.VFC<{
     programContentId?: string
   }>()
   const [previousPage] = useQueryParam('back', StringParam)
+  const [exerciseId] = useQueryParam('exerciseId', StringParam)
+
+  const type = contentCurrentProgress?.programContentType || ''
+  const { loadingSpecificExercise, specificExercise, refetchSpecificExercise, isTaken, exerciseAnswerer } =
+    useSpecificExercise(programContent.id, currentMemberId || '', exerciseId)
+  const {
+    loadingExamId,
+    errorExamId,
+    loadingExam,
+    errorExam,
+    exam,
+    refetch: refetchExam,
+  } = useExam(programContent.id, specificExercise)
+
+  const questionsIdCondition = exam.questions.map(question => {
+    return question.id
+  })
+  const { currentExamData, gaindedPointsTotal, refetchCurrentExamData, questionPointsTotal } =
```

Comment:
in the feature, you just need the exam total score, exam passing score and exercise gained points

you could refer the following information

```typescript
const useExamExercise = (
  programContentId: string,
  memberId: string,
  exerciseId?: string | null
) => {
  const condition: hasura.GET_EXAM_EXERCISEVariables["condition"] = {
    id: exerciseId ? { _eq: exerciseId } : undefined,
    program_content_id: { _eq: programContentId },
    member_id: exerciseId ? undefined : { _eq: memberId },
    answer: { _is_null: false },
  };
  const { loading, error, data } = useQuery<
    hasura.GET_EXAM_EXERCISE,
    hasura.GET_EXAM_EXERCISEVariables
  >(gql`
    query GET_EXAM_EXERCISE($condition: exercise_bool_exp!) {
      exercise(where: $condition) {
        id
        exercise_publics {
          exercise_id
          question_points
        }
        exam {
          id
          passing_score
        }
      }
    }
  `);
};

```